src/main/java/games/strategy/util/Util.java

```diff
@@ -38,19 +40,10 @@
    * @return true if some element in c1 is in c2
    */
   public static <T> boolean someIntersect(final Collection<T> c1, final Collection<T> c2) {
```

Comment:
In theory removing those "return early" checks makes those functions less efficient by increasing the complexity by c1.size()-1 if c2 is empty and c1 is not.
I changed this because linear complexity is ok: "Deleted code is debugged code"